tests/test_run.py

```diff
@@ -0,0 +1,170 @@
+    imagelab = Imagelab(data_path=generate_n_image_files)  # initialize imagelab
+    imagelab.list_default_issue_types()  # list default checks
+
+    ## ==== Test list_default_issue_types() lists all default isssue types====
+    DEFAULT_ISSUE_TYPES = [
+        "dark",
+        "light",
+        "odd_aspect_ratio",
+        "low_information",
+        "exact_duplicates",
+        "near_duplicates",
+        "blurry",
+        "grayscale",
+    ]
+    captured = capsys.readouterr()
+
+    for issue_type in DEFAULT_ISSUE_TYPES:
+        assert issue_type in captured.out
+
+    for issue_type in imagelab._issue_types:
+        assert issue_type in captured.out
+
+    ## ===[TODO] Test visualize produces some result ===
+    # imagelab.visualize()  # visualize random images in dataset
+
+    ## === Test find_issues() finds issues in dataset ===
+    assert len(imagelab.issue_summary) == 0
+    imagelab.find_issues()  # Find issues in the dataset
+    assert len(imagelab.issue_summary) > 0
+
+    ## === Test report() produces print output ===
+    captured = capsys.readouterr()
+    imagelab.report()
+    captured = capsys.readouterr()
+    assert len(captured) > 0
+
+    ## === [TODO] Test visualize works as needed ===
+    # imagelab.visualize(
+    #     issue_types=["blurry"], num_images=8
+    # )  # visualize images that have specific issues
+    #
+    # # Get all images with blurry issue type
+    # blurry_images = imagelab.issues[
+    #     imagelab.issues["is_blurry_issue"] == True
+    #     ].index.to_list()
+    # imagelab.visualize(image_files=blurry_images)  # visualize the given image files
+
+    ## === Test miscellaneous extra information about datasset
+    assert list(imagelab.info.keys()) == [
+        "statistics",
+        "exact_duplicates",
+        "near_duplicates",
+    ]
+    for key in ["color_space", "entropy", "brightness", "blurriness", "aspect_ratio"]:
+        assert key in list(imagelab.info["statistics"].keys())
+
+
+@pytest.mark.usefixtures("generate_n_image_files")
+def test_example2(monkeypatch, generate_n_image_files):
+    monkeypatch.setattr(plt, "show", lambda: None)
+    imagelab = Imagelab(data_path=generate_n_image_files)  # initialize imagelab
+    issue_types = {"near_duplicates": {}}
+    imagelab.find_issues(issue_types)
+    imagelab.report()
+    save_folder = generate_n_image_files + "/T_save_folder/"
+
+    imagelab.save(
+        save_folder
+    )  # optional, just included to show how to save/load this as a file
+
+    # Check for additional types of issues using existing Imagelab
+    imagelab = Imagelab.load(save_folder, generate_n_image_files)
+    issue_types = {"light": {}, "low_information": {}}
+    imagelab.find_issues(issue_types)
+    imagelab.report()
+
+    # Check for an issue with a different threshold
+    issue_types = {"dark": {"threshold": 0.2}}
+    imagelab.find_issues(issue_types)
+    imagelab.report(issue_types=issue_types.keys())  # report only specific issues
```

Comment:
are you planning to add any assert statements here?

Reply:
not in this iteration just trying to get the code covered